src/event.rs

```diff
@@ -173,6 +184,11 @@ impl Default for Mouse {
             column: Column(0),
             cell_side: Side::Left,
             lines_scrolled: 0.0,
+            selection_start_pos: Point {
+                line: Line(0),
+                col: Column(0),
+            },
+            selection_start_side: Side::Left,
```

Comment:
Would it be better to have these be `Option` types? Could that remove the need for a `selection_started` function? I just feel a little weird about these defaults.

Reply:
The defaults are irrelevant but I've chosen to go with "random" defaults over `Option` because it makes accessing them a lot easier.

The `selection_started` method would still be needed unless we clear the `selection_start_side/pos` whenever the selection itself is cleared. I've thought about doing it this way but then decided to go with the current implementation before I've added the `selection_started` method.

I think it might be a viable solution to replace selection start/end with an option and then remove the `selection_started` method, I'll investigate this further just to check if this cleans up the code. Sounds like a good idea, thanks!

Reply:
So after I remembered why I did things the way I did, I wrote about 7 paragraphs explaining why I wasn't able to come up with a simpler solution for this. And after those 7 paragraphs I came up with a simpler solution for this.

So hopefully the changes in my latest commit make things a little better. Let me know if you have any feedback about these changes @nixpulvis!